src/tests/ftest/util/osa_utils.py

```diff
@@ -211,17 +262,38 @@ def ior_thread(self, pool, oclass, test, flags):
         self.ior_cmd.set_daos_params(self.server_group, self.pool)
         self.ior_cmd.dfs_oclass.update(oclass)
         self.ior_cmd.dfs_dir_oclass.update(oclass)
+
+        self.log.info(self.pool_cont_dict)
         # If pool is not in the dictionary,
-        # initialize its container as None.
+        # initialize its container list to None
+        # {poolA : [None, None], [None, None]}
         if self.pool not in self.pool_cont_dict:
-            self.pool_cont_dict[self.pool] = None
+            self.pool_cont_dict[self.pool] = [None] * 4
         # Create container if the pool doesn't have one.
         # Otherwise, use the existing container in the pool.
-        if self.pool_cont_dict[self.pool] is None:
-            self.add_container(self.pool)
-            self.pool_cont_dict[self.pool] = self.container
+        # pool_cont_dict {pool A: [containerA, Updated,
+        #                          containerB, Updated],
+        #                 pool B : containerA, Updated,
+        #                          containerB, None]}
```

Comment:
It would be great if you could define this in the class description or somewhere easy to find. I didn't understand what delete_extra_container is doing until I see this comment.

Reply:
Thanks Makito... Yes, I am planning to take out the entire if condition logic into a separate method... In this way, ior_thread can used for single and multi-container read/write envs.